app/controllers/users.js

```diff
@@ -79,17 +79,42 @@ exports.sendInviteEmail = (req, res) => {
  * Auth callback
  */
 
+const getJWT = (tokenInfo, jwtSecret) => new Promise((resolve, reject) => {
+  if (tokenInfo) {
+    jwt.sign(
+      tokenInfo, jwtSecret,
+      (tokenError, generatedToken) => {
+        if (tokenError) {
+          reject(new Error(tokenError));
+        } else if (generatedToken) {
+          resolve(generatedToken);
+        } else {
+          reject(new Error('Something went wrong'));
+        }
+      }
+    );
+  } else {
+    reject(new Error('No Information Supplied'));
+  }
+});
 
-exports.authCallback = (req, res, next) => {
-  res.redirect('/chooseavatars');
+exports.authCallback = (req, res) => {
+  getJWT(req.user.name, process.env.JWT_SECRET)
+    .then((token) => {
+      res.cookie('cfhToken', token);
```

Comment:
Nitpick: Why are you using res.cookie

Reply:
I used res.cookie because I needed to pass the token to the frontend where localStorage is available. The token in the cookie is destroyed afterwards

Reply:
Alright. Setting it in the response header is another option

Reply:
headers are usually short lived. An instance when a user refreshes the app, the token in the header will be lost